// @metaplex-foundation/digital-asset-standard-api/dist/src/decorator.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createDasApiDecorator = void 0;
const errors_1 = require("./errors");
const createDasApiDecorator = (rpc) => ({
    ...rpc,
    getAsset: async (assetId) => {
        const asset = await rpc.call('getAsset', [assetId]);
        if (!asset)
            throw new errors_1.DasApiError(`Asset not found: ${assetId}`);
        return asset;
    },
    getAssetProof: async (assetId) => {
        const proof = await rpc.call('getAssetProof', [assetId]);
        if (!proof)
            throw new errors_1.DasApiError(`No proof found for asset: ${assetId}`);
        return proof;
    },
    getAssetsByAuthority: async (input) => {
        if (typeof input.page === 'number' && (input.before || input.after)) {
            throw new errors_1.DasApiError('Pagination Error. Please use either page or before/after, but not both.');
        }
        const assetList = await rpc.call('getAssetsByAuthority', [
            input.authority,
            input.sortBy ?? null,
            input.limit ?? null,
            input.page ?? 1,
            input.before ?? null,
            input.after ?? null,
        ]);
        if (!assetList) {
            throw new errors_1.DasApiError(`No assets found for authority: ${input.authority}`);
        }
        return assetList;
    },
    getAssetsByCreator: async (input) => {
        if (typeof input.page === 'number' && (input.before || input.after)) {
            throw new errors_1.DasApiError('Pagination Error. Please use either page or before/after, but not both.');
        }
        const assetList = await rpc.call('getAssetsByCreator', [
            input.creator,
            input.onlyVerified,
            input.sortBy ?? null,
            input.limit ?? null,
            input.page ?? 1,
            input.before ?? null,
            input.after ?? null,
        ]);
        if (!assetList) {
            throw new errors_1.DasApiError(`No assets found for creator: ${input.creator}`);
        }
        return assetList;
    },
    getAssetsByGroup: async (input) => {
        if (typeof input.page === 'number' && (input.before || input.after)) {
            throw new errors_1.DasApiError('Pagination Error. Please use either page or before/after, but not both.');
        }
        const assetList = await rpc.call('getAssetsByGroup', [
            input.groupKey,
            input.groupValue,
            input.sortBy ?? null,
            input.limit ?? null,
            input.page ?? 1,
            input.before ?? null,
            input.after ?? null,
        ]);
        if (!assetList) {
            throw new errors_1.DasApiError(`No assets found for group: ${input.groupKey} => ${input.groupValue}`);
        }
        return assetList;
    },
    getAssetsByOwner: async (input) => {
        if (typeof input.page === 'number' && (input.before || input.after)) {
            throw new errors_1.DasApiError('Pagination Error. Please use either page or before/after, but not both.');
        }
        const assetList = await rpc.call('getAssetsByOwner', [
            input.owner,
            input.sortBy ?? null,
            input.limit ?? null,
            input.page ?? 1,
            input.before ?? null,
            input.after ?? null,
        ]);
        if (!assetList) {
            throw new errors_1.DasApiError(`No assets found for owner: ${input.owner}`);
        }
        return assetList;
    },
    searchAssets: async (input) => {
        if (typeof input.page === 'number' && (input.before || input.after)) {
            throw new errors_1.DasApiError('Pagination Error. Please use either page or before/after, but not both.');
        }
        const assetList = await rpc.call('searchAssets', [
            input.negate ?? null,
            input.conditionType ?? null,
            input.interface ?? null,
            input.owner ?? null,
            input.ownerType ?? null,
            input.creator ?? null,
            input.creatorVerified ?? null,
            input.authority ?? null,
            input.grouping ?? null,
            input.delegate ?? null,
            input.frozen ?? null,
            input.supply ?? null,
            input.supplyMint ?? null,
            input.compressed ?? null,
            input.compressible ?? null,
            input.royaltyModel ?? null,
            input.royaltyTarget ?? null,
            input.royaltyAmount ?? null,
            input.burnt ?? null,
            input.sortBy ?? null,
            input.limit ?? null,
            input.page ?? 1,
            input.before ?? null,
            input.after ?? null,
            input.jsonUri ?? null,
        ]);
        if (!assetList) {
            throw new errors_1.DasApiError('No assets found for the given search criteria');
        }
        return assetList;
    },
});
exports.createDasApiDecorator = createDasApiDecorator;
